src/util-inl.h

```diff
@@ -198,6 +198,18 @@ TypeName* Unwrap(v8::Local<v8::Object> object) {
   return static_cast<TypeName*>(pointer);
 }
 
+void SwapBytes(uint16_t* dst, const uint16_t* src, size_t buflen) {
+  for (size_t i = 0; i < buflen; i++) {
+      #ifdef __GNUC__
+        dst[i] = __builtin_bswap16(src[i]);
```

Comment:
it really depends on what optimization level is chosen and what compiler is using. gcc 4.8 with -O1 is not be able to lowering the shifting operation to a swap. Here is the example:
```
#include <stdio.h>

void test1(unsigned short *a) {
    *a = (*a >> 8) | (*a << 8);
}

void test2(unsigned short *b) {
    *b = __builtin_bswap16(*b);
}

int main() {
    short a = 4;
    short b = 5;
    test1(&a);
    test2(&b);
    printf("%d %d", (int)a, (int)b);
}
```
compile with "gcc -O1 --no-inline test.c" on a PPC machine
```
(gdb) disas test1
Dump of assembler code for function .test1:
   0x0000000010000604 <+0>:     lhz     r9,0(r3)
   0x0000000010000608 <+4>:     rlwinm  r10,r9,8,0,23
   0x000000001000060c <+8>:     rlwinm  r9,r9,24,8,31
   0x0000000010000610 <+12>:    or      r9,r10,r9
   0x0000000010000614 <+16>:    sth     r9,0(r3)
   0x0000000010000618 <+20>:    blr
   0x000000001000061c <+24>:    .long 0x0
   0x0000000010000620 <+28>:    .long 0x0
   0x0000000010000624 <+32>:    .long 0x0
End of assembler dump.
(gdb) disas test2
Dump of assembler code for function .test2:
   0x0000000010000628 <+0>:     lhz     r9,0(r3)
   0x000000001000062c <+4>:     sthbrx  r9,0,r3
   0x0000000010000630 <+8>:     blr
   0x0000000010000634 <+12>:    .long 0x0
   0x0000000010000638 <+16>:    .long 0x0
   0x000000001000063c <+20>:    .long 0x0
End of assembler dump.

```